apps/dashr-tsne/app.R

```diff
@@ -0,0 +1,1195 @@
+library(dplyr)
+library(dash)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(data.table)
+library(plotly)
+library(rlist)
+library(glue)
+library(parallel)
+library(compiler)
+library(dtplyr)
+library(knitr)
+library(png)
+library(magick)
+library(base64enc)
+library(magrittr)
+library(Rtsne)
+library(stringr)
+library(jsonlite)
+library(tictoc)
+
+appName <- Sys.getenv("DASH_APP_NAME")
+pathPrefix <- sprintf("/%s/", appName)
+
+
+Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+           DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+setwd(sprintf("/app/apps/%s", appName))
+
+l. <- cmpfun(list)
+
+IMAGE_DATASETS <- c("mnist_3000", "cifar_gray_3000", "fashion_3000")
+WORD_EMBEDDINGS <- c("wikipedia_3000", "twitter_3000", "crawler_3000")
+
+
+readme <- htmlDiv(children=l.(
+  htmlButton("Close", id='close-button', n_clicks=0)
+  ,
```

Comment:
This is more of a style issue -- I tend to keep function calls in one line if possible (so that I could  select the function easily by double-clicking in order copy/paste to test each of them). Will make the change if needed.

Reply:
as in separating the comma and the function call*